--- boost_1_65_0-orig/boost/spirit/home/classic/core/non_terminal/impl/grammar.ipp	2017-08-19 16:49:48.000000000 +0000
+++ boost_1_65_0/boost/spirit/home/classic/core/non_terminal/impl/grammar.ipp	2017-09-06 23:51:46.339124628 +0000
@@ -13,10 +13,16 @@
 
 #if !defined(BOOST_SPIRIT_SINGLE_GRAMMAR_INSTANCE)
 #include <boost/spirit/home/classic/core/non_terminal/impl/object_with_id.ipp>
+#include <boost/smart_ptr/detail/sp_disable_deprecated.hpp>
 #include <algorithm>
 #include <functional>
 #include <boost/move/unique_ptr.hpp>
 #include <boost/weak_ptr.hpp>
+
+#if defined( BOOST_SP_DISABLE_DEPRECATED )
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
 #endif
 
 #ifdef BOOST_SPIRIT_THREADSAFE
@@ -362,4 +368,10 @@
 
 }} // namespace boost::spirit
 
+#if !defined(BOOST_SPIRIT_SINGLE_GRAMMAR_INSTANCE)
+#if defined( BOOST_SP_DISABLE_DEPRECATED )
+#pragma GCC diagnostic pop
+#endif
+#endif
+
 #endif
