--- boost_1_65_0-orig/boost/asio/detail/socket_types.hpp	2017-08-19 16:49:39.000000000 +0000
+++ boost_1_65_0/boost/asio/detail/socket_types.hpp	2017-09-06 23:58:08.163129288 +0000
@@ -62,7 +62,7 @@
    || defined(__OpenBSD__) || defined(__linux__)
 #  include <poll.h>
 # elif !defined(__SYMBIAN32__)
-#  include <sys/poll.h>
+#  include <poll.h>
 # endif
 # include <sys/types.h>
 # include <sys/stat.h>
